ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -273,13 +275,16 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
           stmt -> buildSingleQueryAst(
               stmt.getStatement(), stmt.getStatementText(), tempMetaStore, tempMetaStoreForParser));
 
-      return statements
-          .stream()
-          .filter(Objects::nonNull)
-          .collect(Collectors.toList());
-
-    } catch (final Exception e) {
-      throw new ParseFailedException("Exception while processing statements :" + e.getMessage(), e);
+      if (queriesOnly) {
```

Comment:
This smells... looking at the production code, the only place that passes in `false` for `queriesOnly` is `StandaloneExecutor.executeStatements` and it would blow up with an NPE if any of the elements in the returned list was null. 

Also, `queriesOnly` doesn't limit the method to only returning queries, i.e. instances of `Query`.

Therefore, I'd suggest dropping the `queriesOnly` and _always_ filtering for nulls. Then just fix up the tests.